hardware/esp8266com/esp8266/libraries/ESP8266HTTPUpdateServer/src/ESP8266HTTPUpdateServer.cpp

```diff
@@ -0,0 +1,88 @@
+#include <Arduino.h>
+#include <WifiClient.h>
+#include <WifiServer.h>
+#include <ESP8266WebServer.h>
+#include <WifiUDP.h>
+#include "ESP8266HTTPUpdateServer.h"
+
+
+const char* ESP8266HTTPUpdateServer::_serverIndex = "<html><body><form method='POST' action='/update' enctype='multipart/form-data'>"
+                                          "<input type='file' name='update'>"
+                                          "<input type='submit' value='Update'>"
+                                      "</form>"
+                                "</body></html>";
+
+ESP8266HTTPUpdateServer::ESP8266HTTPUpdateServer(int port, bool serial_debug)
+{
+  _serial_output = serial_debug;
+  _server = NULL;
+  _port = port;
+}
+
+ESP8266HTTPUpdateServer::~ESP8266HTTPUpdateServer()
+{
+  if (_server)
+    delete _server;
```

Comment:
Good catch, also like this, we make the dependency explicit, and make sure the user will import the ESP8266WebServer from the main .ino, otherwise, it won't get included/linked since arduino libs can't set dependencies AFAIK.